src/client/proxy/handlers/common.ts

```diff
@@ -133,25 +134,28 @@ const getHttpRequestHandler = () => async (req: FastifyRequest, res: FastifyRepl
 
                 const renderer = new Renderer({localDir: publicPath} as any);
 
-                res.header('Content-Type', 'text/html');
+                const contentType = ext ? getContentTypeFromExt(ext) : 'text/html';
+                res.header('Content-Type', contentType);
+
                 // TODO: sanitize
-                return renderer.render(templateFilename, templateFileContents, host, {
+                return await renderer.render(templateFilename, templateFileContents, host, {
```

Comment:
`await` is redundant when returning from `async` function.
BTW I think we can add the rule for it: https://eslint.org/docs/latest/rules/no-return-await#:~:text=Using%20return%20await%20inside%20an,before%20resolving%20the%20outer%20Promise.

Reply:
Actually, that was exactly the problem with error propagation.
Skipping await here leads to all errors from e.g. ZHTML renderer being thrown higher up instead of here on the spot, and they don't end up being caught with the local try...catch which returns the error message to the browser